### PR TITLE
additional gap conductance tests

### LINKED ISSUE #6161: Expanded options for cylindrical/spherical gap geometry
We currently have an option for thermal contact to make appropriate modifications to the gap conductance when it is known that the gap is cylindrical or spherical. This allows gap conductance to be used with large gaps on these types of geometry.

The current implementation only allows these options to be used with axisymmetric or spherical coordinate system models.  They are also applicable to higher dimensionality models when they have this type of geometry.  Add options to allow the user to ask for these formulations in those cases.

### PR DESCRIPTION
**Pull request by novasr**
closes #6161

### CODE REVIEW

File: modules/combined/tests/gap_heat_transfer_htonly/tests [hunk 1]
```diff
@@ -31,4 +31,24 @@
     compiler = 'CLANG GCC'
   [../]
 
+  [./cyl3D]
+    type = Exodiff
+    input = 'cyl3D.i'
+    exodiff = 'cyl3D_out.e'
+  [../]
```
Comment (permcody) [inline modules/combined/tests/gap_heat_transfer_htonly/tests]: Each of these tests needs a `superlu = true` line for people who don't have that package available

**Comment by novasr** [comment]
There are only four new files in gold/. Where are the other two?

**Comment by permcody** [comment]
My bad, my statement is correct but of course some of those exodus files are input meshes so ignore it.

**Comment by novasr** [comment]
Ok. Made the change to tests.

**Comment by permcody** [comment]
:+1:

**Comment by permcody** [comment]
Bad News: All of your new tests fail on Intel with fairly significant diffs (above single precision values). You'll either need to look at these tests more or skip them on Intel. I still feel something has got to be wrong for those test to be that far off.

https://www.moosebuild.org/job/10515/

**Comment by permcody** [comment]
@novasr - What's the plan here? Are we going to skip these tests again? Reworking them? They are showing up on our daily testing now and failing every time!

**Comment by novasr** [comment]
Skip for now.

**Pull request merged**